api/http/proxy/factory/kubernetes/transport.go

```diff
@@ -79,6 +77,17 @@ func (transport *baseTransport) executeKubernetesRequest(request *http.Request)
 
 	resp, err := transport.httpTransport.RoundTrip(request)
 
+	if err == nil && resp.StatusCode == http.StatusMovedPermanently {
```

Comment:
- should it be applicable to only 301 redirects or any 3xx?
- could we please add a comment or even better a test to explain what manipulations do we do with the Location URL and why?